zebra-consensus/src/block/subsidy/funding_streams.rs

```diff
@@ -50,3 +48,116 @@ pub fn funding_stream_values(
     }
     Ok(results)
 }
+
+/// Returns the minumum height after the first halving
+/// as described in [protocol specification §7.10][7.10]
+///
+/// [7.10]: https://zips.z.cash/protocol/protocol.pdf#fundingstreams
+fn height_for_halving(network: Network) -> Height {
+    // First halving on Mainnet is at Canopy
+    // while in Testnet is at block 1_116_000
+    // https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+    match network {
+        Network::Mainnet => Canopy
+            .activation_height(network)
+            .expect("canopy activation height should be available"),
+        Network::Testnet => Height(1_116_000),
```

Comment:
Can we calculate this value, or turn it into a constant?

Reply:
I am not sure if we can calculate the value, probably i guess but unsure if it will worth to add more complexity. I moved the block height value into a constant in https://github.com/ZcashFoundation/zebra/pull/3040/commits/4e486e9c162c45250c42c6675680f24492ade5c0